draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -2797,11 +2797,14 @@ Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
 
 <section title="Identifying HTTPS Servers" anchor="https.identify.server">
 <t>
-   In general, HTTP/TLS requests are generated by dereferencing a URI.
-   As a consequence, the hostname for the server is known to the client.
-   If the hostname is available, the client &MUST; check it against the
-   server's identity, as presented in the server's Certificate message,
-   to prevent an on-path attacker from substituting the key.
+   In general, HTTP/TLS requests are generated by dereferencing a URI. As a
+   consequence, the hostname for the server is known to the client. If the
+   hostname is available, the client &MUST; use the hostname as a reference
+   identity as input to the verification processed defined in <xref
+   target="RFC6125" x:fmt="of" x:sec="6"/>. This process requires that a client be configured with a set of
+   trust anchors. Certificate verification is used to prevent server
+   impersonation by an on-path attacker or by an attacker that controls name
+   resolution.
```

Comment:
```suggestion
   When HTTP/TLS requests are generated by dereferencing a URI which
   identifies an origin server within its authority component,
   the client &MUST; use the origin server identity from the URI as the
   reference identity for the verification process defined in <xref
   target="RFC6125" x:fmt="of" x:sec="6"/>.
   This process requires that a client be configured with a set of
   trust anchors. Certificate verification is used to prevent server
   impersonation by an on-path attacker or by an attacker that controls name
   resolution.
```
This makes dereferencing a URI the condition for this MUST, since there are uses of HTTP/TLS beyond that.